### ISSUE TITLE
[gogs] Incorrect resolution of relative "avatar_url"

### ISSUE THREAD

**Issue by strk** [drone/drone#1701]
Drone is showing all broken images as avatars for repositories from Gogs source.
I've seen it's due to relative URLs being found in "avatar_url" element of the
``api/v1/orgs/orgname`` endpoint, for example:
``"avatar_url":"/gogs/avatars/6"``

Drone is turning that into an url relative to the Drone url, not the Gogs url.

**Comment by bradrydzewski** [comment]
I am currently unable to repeat.

We see a heavy amount of variation in gogs URLs and have a function to fix / normalize urls:
https://github.com/drone/drone/blob/master/remote/gogs/helper.go#L108

I'm using gogs `0.9.35.0702` and do not have any issues with avatars, however, I realize there are multiple avatar configurations. Please feel free to send a pull request if you think the avatar code can be improved: https://github.com/drone/drone/blob/master/remote/gogs/helper.go#L108

**Comment by strk** [comment]
@bradrydzewski it might be only for organizations, not real users.

**Comment by strk** [comment]
For an example, see https://git.osgeo.org/gogs/api/v1/orgs/postgis

**Comment by strk** [comment]
Gogs 0.9.27.0512 here (for the record) -- and drone 0.4 (could have been fixed in 0.5)

**Comment by strk** [comment]
In ExpandAvatars from helpers.go:

```
        if !strings.HasPrefix(rawurl, "/avatars/") {
          return rawurl
        }
```

The problem is that my Gogs instance runs from a subdir, so the prefix is actually `/gogs/avatars`, making the ExpandAvatar function return the rawurl w/out parsing it

**Comment by strk** [comment]
interesting, commenting out that snippet fixes all the avatars previously broken, but breaks the only one that worked (an absolute url)

**Comment by strk** [comment]
Ok I have a working fix, fine to adapt the tests with this ?
```
    "http://1.gravatar.com/avatar/8c58a0be77ee441bb8f8595b7f1b4e87" does not equal "//1.gravatar.com/avatar/8c58a0be77ee441bb8f8595b7f1b4e87"
```

That is, current tests expect '//' prefix, while my code makes it 'http://' (based on repo url, using core net/url)

**Comment by strk** [comment]
This would be fixed by #1703 (if you want to reopen until that one is merged)
